balances/src/lib.rs

```diff
@@ -99,6 +99,14 @@ decl_error! {
 }
 
 impl<T: Config> Module<T> {
+	pub fn get_balances(account: &T::AccountId) -> Vec<(CommunityIdentifier, BalanceType)> {
+		let mut balances: Vec<(CommunityIdentifier, BalanceType)> = vec![];
+		for community in <encointer_communities::Module<T>>::community_identifiers().iter() {
+			balances.push((*community, Balance::<T>::get(community, account.clone()).principal));
```

Comment:
it's not ok to just push the principal
The principal always comes with its individual block height and you need both informations to apply demurrage. As demurrage can differ among communities, I think it would actually be better to apply demurrage directly in this query. It's not such an expensive computation

Reply:
and if we return the current balance with demurrage applied, we have to return the block-number for which this applies as well, otherwise, the UI can't update the balance independently over time. As the UI needs to be capable to apply demurrage anyway, we should consider returning values without applying demurrage. 

To summarize the result of my zigzagging: We should return the entire `BalanceEntry` instead of just the principal  `BalanceType` and all is good.